format/fit/fit.go

```diff
@@ -216,13 +236,24 @@ func fitDecodeDefinitionMessage(d *decode.D, drc *dataRecordContext, lmfd localF
 	}
 }
 
-func fieldUint(fieldFn func(string, ...scalar.UintMapper) uint64, expectedSize uint64, fDef mappers.LocalFieldDef, uintFormatter scalar.UintFn, fdc *fileDescriptionContext, valMap valueMap) {
+func readUint(d *decode.D, fDef mappers.LocalFieldDef, valMap valueMap) {
+	expectedSize := expectedSizeMap[fDef.Type]
+	if fDef.Size != expectedSize {
+		d.SeekRel(int64(fDef.Size) * 8) // skip over array types since they cannot be referenced by subfields
```

Comment:
How come this seek and the a seek in `fitDecodeDataMessage` does not cause "gaps", some other field cover that bit range?

Reply:
This seek and the one on line 360 are just to keep the pos in sync since we only care about uint fields on the first pass. On line 363 we seek back to before the pre-read and read all the data with d.Field-methods.